app/database.py

```diff
@@ -57,4 +57,29 @@ def createNewChallenge(creator_id, title, path, dt_submit, dt_finish):
         dt_finish=dt_finish,
         finished=False)
     db.session.add(newChallenge)
-    db.session.commit()
\ No newline at end of file
+    db.session.commit()
+
+# Given a Challenge model, find the submission with the fewest articles in its path (the winner of the challenge)
+def findWinner(challenge):
+    maxArticles = 51 # For what we currently have, each path has a maximum length of 50 articles
+    winningSubmission = None
+    for submission in challenge.attempts:
+        # If there's a tie, the earliest submission should be the winner, and in the database, it automatically sorts via attempt submission
+        if submission.article_no < maxArticles:
+            maxArticles = submission.article_no
+            winningSubmission = submission.id
+    return winningSubmission
+
+# Query the database to see if a challenge is finished. If so, update the database to store who won the challenge.
+def updateDB():
+    challenges = Challenge.query.all()
+    for challenge in challenges:
+        # challenge.finished == False is most likely bad practice
+        if ((challenge.dt_finish-int(time.time())) <= 0 and challenge.finished == False):
```

Comment:
is challenge.finished necessary here, even? could we not remove it from the table entirely?

Reply:
Although i see why (it could slow things down with many database calls). I'm fine to keep it like this for now and just make a note to look over it later

Reply:
When I made the database I thought it would be much simpler for us in the future if we only had to check `challenge.finished` rather than `challenge.dt_finish-int(time.time())`

Reply:
Yeah, definitely agree there.